pkg/controller/razeedeployment/razeedeployment_controller.go

```diff
@@ -856,275 +871,308 @@ func (r *ReconcileRazeeDeployment) Reconcile(request reconcile.Request) (reconci
+	instance.Status.Conditions.SetCondition(status.Condition{
+		Type:    marketplacev1alpha1.ConditionInstalling,
+		Status:  corev1.ConditionTrue,
+		Reason:  marketplacev1alpha1.ReasonRhmRemoteResourceS3DeploymentInstalled,
+		Message: message,
+	})
+
+	watchKeeperDeployment := &appsv1.Deployment{}
+	reqLogger.V(0).Info("Finding watch-keeper deployment")
+	err = r.client.Get(context.TODO(), types.NamespacedName{
+		Name:      utils.RHM_WATCHKEEPER_DEPLOYMENT_NAME,
+		Namespace: request.Namespace,
+	}, watchKeeperDeployment)
+	if errors.IsNotFound(err) {
+		reqLogger.V(0).Info("Creating watch-keeper deployment")
+		watchKeeperDeployment = r.makeWatchKeeperDeployment(instance)
+		err = r.client.Create(context.TODO(), watchKeeperDeployment)
 		if err != nil {
-			reqLogger.Error(err, "Failed to update Job(s) from Cluster")
+			reqLogger.Error(err, "Failed to create watch-keeper deployment on cluster")
 			return reconcile.Result{}, err
 		}
-		return reconcile.Result{Requeue: true}, nil
-	}
+		reqLogger.Info("watch-keeper deployment created successfully")
 
-	if !reflect.DeepEqual(instance.Status.JobState, foundJob.Status) {
-		// Update status and conditions
-		instance.Status.JobState = foundJob.Status
-		for _, jobCondition := range foundJob.Status.Conditions {
-			instance.Status.JobConditions = &jobCondition
-		}
+		message := "watch-keeper install starting"
+		instance.Status.Conditions.SetCondition(status.Condition{
+			Type:    marketplacev1alpha1.ConditionInstalling,
+			Status:  corev1.ConditionTrue,
+			Reason:  marketplacev1alpha1.ReasonWatchKeeperDeploymentStart,
+			Message: message,
+		})
 
-		instance.Status.RazeeJobInstall = &marketplacev1alpha1.RazeeJobInstallStruct{
-			RazeeNamespace:  secretName,
-			RazeeInstallURL: instance.Spec.DeployConfig.FileSourceURL,
-		}
+		_ = r.client.Status().Update(context.TODO(), instance)
 
-		err = r.client.Status().Update(context.TODO(), instance)
-		if err != nil {
-			reqLogger.Error(err, "Failed to update JobState")
-			return reconcile.Result{}, err
-		}
-		reqLogger.V(0).Info("Updated JobState")
 		return reconcile.Result{Requeue: true}, nil
-	}
 
-	if foundJob.Status.Succeeded != 1 {
-		reqLogger.Info("waiting for job to finish")
-		return reconcile.Result{RequeueAfter: time.Second * 15}, nil
+	} else if err != nil {
+		reqLogger.Error(err, "Failed to get RemoteResourceS3 from Cluster")
+		return reconcile.Result{}, err
 	}
 
-	// if the job succeeds apply the parentRRS3 and patch the Infrastructure, ClusterVersion and Console resources
-	if foundJob.Status.Succeeded == 1 {
-		parentRRS3 := &unstructured.Unstructured{}
-		parentRRS3.SetGroupVersionKind(schema.GroupVersionKind{
-			Group:   "deploy.razee.io",
-			Kind:    "RemoteResourceS3",
-			Version: "v1alpha2",
-		})
+	if err := controllerutil.SetControllerReference(instance, watchKeeperDeployment, r.scheme); err != nil {
+		reqLogger.Error(err, "Failed to set controller reference")
+		return reconcile.Result{}, err
+	}
 
```

Comment:
Please check for changes to the watch-keeper deployment and update if necessary. In particular the version may change in the deployment.